src/internal/connector/data_collections.go

```diff
@@ -51,50 +51,48 @@ func (gc *GraphConnector) DataCollections(ctx context.Context, sels selectors.Se
 
 		return colls, nil
 	default:
-		return nil, errors.Errorf("service %s not supported", sels)
+		return nil, errors.Errorf("service %s not supported", sels.Service.String())
 	}
 }
 
-func verifyBackupInputs(sel selectors.Selector, mapOfUsers map[string]string) error {
-	var personnel []string
+func verifyBackupInputs(sels selectors.Selector, userPNs, siteIDs []string) error {
+	var ids []string
 
-	// retrieve users from selectors
-	switch sel.Service {
-	case selectors.ServiceExchange:
-		backup, err := sel.ToExchangeBackup()
-		if err != nil {
-			return err
-		}
+	resourceOwners, err := sels.ResourceOwners()
+	if err != nil {
+		return errors.Wrap(err, "invalid backup inputs")
+	}
 
-		for _, scope := range backup.Scopes() {
-			temp := scope.Get(selectors.ExchangeUser)
-			personnel = append(personnel, temp...)
-		}
-	case selectors.ServiceOneDrive:
-		backup, err := sel.ToOneDriveBackup()
-		if err != nil {
-			return err
-		}
+	switch sels.Service {
+	case selectors.ServiceExchange, selectors.ServiceOneDrive:
+		ids = userPNs
 
-		for _, user := range backup.Scopes() {
-			temp := user.Get(selectors.OneDriveUser)
-			personnel = append(personnel, temp...)
-		}
+	case selectors.ServiceSharePoint:
+		ids = siteIDs
+	}
 
-	default:
-		return errors.New("service %s not supported")
+	// verify resourceOwners
+	normROs := map[string]struct{}{}
+
+	for _, id := range ids {
+		normROs[strings.ToLower(id)] = struct{}{}
 	}
 
-	// verify personnel
-	normUsers := map[string]struct{}{}
+	for _, ro := range resourceOwners.Includes {
+		if _, ok := normROs[strings.ToLower(ro)]; !ok {
+			return fmt.Errorf("included resource owner %s not found within tenant", ro)
+		}
+	}
 
-	for k := range mapOfUsers {
-		normUsers[strings.ToLower(k)] = struct{}{}
+	for _, ro := range resourceOwners.Excludes {
+		if _, ok := normROs[strings.ToLower(ro)]; !ok {
+			return fmt.Errorf("excluded resource owner %s not found within tenant", ro)
+		}
```

Comment:
this may be a little more strict than we need it tbh. If we're excluding the resource owner I don't really think it makes a difference if we find it or not. The only thing that matters is that we don't include it in the backup.

For example, this could be a set of archived users that no longer need backed up from people who have left the company. After a while those users may be deleted, but whatever is triggering the backups should still work and ideally I would think it would be alright to not fail if the script triggering the backups wasn't updated immediately to reflect the fact that users were deleted

It may be good to check with someone more in tune with UX than me though heh